python/orca/src/bigdl/orca/learn/tf2/pyspark_estimator.py

```diff
@@ -151,34 +152,36 @@ def transform_func(iter, init_param, param):
                     param["validation_data_creator"] = make_data_creator(valid_list)
                     return SparkRunner(**init_param).step(**param)
 
-                res = data.zip(validation_data).rdd.repartition(self.num_workers).barrier() \
+                self.states_rdd = data.zip(validation_data).rdd.repartition(self.num_workers).barrier() \
```

Comment:
I think there is an assumption of this line that `data` and `validataion_data` has the same number of partitions and the same number of elements on each partition.
I think the assumption may not hold if `OrcaContext._shard_size` is not None.